src/main/scala/de/upb/cs/swt/delphi/crawler/Configuration.scala

```diff
@@ -35,6 +58,19 @@ class Configuration  {
   val elasticActorPoolSize : Int = 4
   val callGraphStreamPoolSize : Int = 4
 
+  val instanceName = "MyCrawlerInstance"
+  val instanceRegistryUri : String = sys.env.getOrElse("DELPHI_IR_URI", "http://localhost:8085")
+
+  lazy val usingInstanceRegistry : Boolean = assignedID match {
+    case Some(_) => true
+    case None => false
+  }
+
+  lazy val assignedID : Option[Long] = InstanceRegistry.register(this) match {
```

Comment:
Do you really want to lazyly register at the IR and not explicitly at the startup of the crawler?

Reply:
Since the crawler has its preflight-checks in place i can be sure that the `ElasticReachablePreflightCheck` will resolve the respective lazy values during the startup routine. The main reason for this construct was that i needed to access the ID in a way that does not execute `register` multiple times, while also not using a `var`. Can you think of a better way to handle this ?

Reply:
Ahhhh okay... Yes, then we'll keep it this way. I haven't thought of that.